aspnetcore/fundamentals/servers/kestrel/endpoints.md

```diff
@@ -5,144 +5,235 @@ description: Learn about configuring endpoints with Kestrel, the cross-platform
-* `UseHttps(Action<HttpsConnectionAdapterOptions> configureOptions)`
+* <xref:Microsoft.AspNetCore.Server.Kestrel.Core.KestrelServerOptions.Listen%2A>
+* <xref:Microsoft.AspNetCore.Server.Kestrel.Core.KestrelServerOptions.ListenLocalhost%2A>
+* <xref:Microsoft.AspNetCore.Server.Kestrel.Core.KestrelServerOptions.ListenAnyIP%2A>
+* <xref:Microsoft.AspNetCore.Server.Kestrel.Core.KestrelServerOptions.ListenUnixSocket%2A>
+* <xref:Microsoft.AspNetCore.Server.Kestrel.Core.KestrelServerOptions.ListenNamedPipe%2A>
 
-`ListenOptions.UseHttps` parameters:
+When both the `Listen` and [UseUrls](#configure-endpoints-with-urls) APIs are used simultaneously, the `Listen` endpoints override the `UseUrls` endpoints.
 
-* `filename` is the path and file name of a certificate file, relative to the directory that contains the app's content files.
-* `password` is the password required to access the X.509 certificate data.
-* `configureOptions` is an `Action` to configure the `HttpsConnectionAdapterOptions`. Returns the `ListenOptions`.
-* `storeName` is the certificate store from which to load the certificate.
-* `subject` is the subject name for the certificate.
-* `allowInvalid` indicates if invalid certificates should be considered, such as self-signed certificates.
-* `location` is the store location to load the certificate from.
-* `serverCertificate` is the X.509 certificate.
+#### Bind to a TCP socket
 
-In production, HTTPS must be explicitly configured. At a minimum, a default certificate must be provided.
+The <xref:Microsoft.AspNetCore.Server.Kestrel.Core.KestrelServerOptions.Listen%2A>, <xref:Microsoft.AspNetCore.Server.Kestrel.Core.KestrelServerOptions.ListenLocalhost%2A>, and <xref:Microsoft.AspNetCore.Server.Kestrel.Core.KestrelServerOptions.ListenAnyIP%2A> methods bind to a TCP socket:
 
-Supported configurations described next:
+:::code language="csharp" source="~/fundamentals/servers/kestrel/samples/6.x/KestrelSample/Snippets/Program.cs" id="snippet_Listen":::
+
+The preceding example:
+
+* Configures endpoints that listen on port 5000 and 5001.
+* Configures HTTPS for an endpoint with the <xref:Microsoft.AspNetCore.Hosting.ListenOptionsHttpsExtensions.UseHttps%2A> extension method on <xref:Microsoft.AspNetCore.Server.Kestrel.Core.ListenOptions>. For more information, see [Configure HTTPS in code](#configure-https-in-code).
+
+[!INCLUDE [How to make an X.509 cert](~/includes/make-x509-cert.md)]
 
-* No configuration
-* Replace the default certificate from configuration
-* Change the defaults in code
+#### Bind to a Unix socket
 
-### No configuration
+Listen on a Unix socket with <xref:Microsoft.AspNetCore.Server.Kestrel.Core.KestrelServerOptions.ListenUnixSocket%2A> for improved performance with Nginx, as shown in this example:
+
+:::code language="csharp" source="~/fundamentals/servers/kestrel/samples/6.x/KestrelSample/Snippets/Program.cs" id="snippet_ListenUnixSocket":::
```

Comment:
The example shows http and https using the same sock? How would that work?

Reply:
I believe the second call will error. I'll remove the https line.

That mistake must have been there for years.